### PR TITLE
Adds workflow caching of go modules and binaries

### PR DESCRIPTION
**Pull request by codefromthecrypt**

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
I don't think we'll get to see the impact of this cache until it is merged.

File: .bingo/README.md [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+The contents here were generated by [bingo](https://github.com/bwplotka/bingo) but have no runtime
+dependency on it in any way.
 
-## Requirements
+Integration in [Makefile](../Makefile) requires inclusion of the [Variables.mk](Variables.mk), then
+reference as a variable like `$(GOIMPORTS)`.
 
-* Go 1.14+
+This differs slightly from the defaults of the Bingo project:
+* ENV variable support is not included as this project only uses `make`
```
Comment (codefromthecrypt) [inline .bingo/README.md:13]: I removed this and rewrote the README as it I think the result is more relevant than the generated README

File: .github/workflows/commit.yaml [hunk 2]
```diff
@@ -18,23 +18,32 @@ on:
...
         with:
-          go-version: '1.16.3'
+          go-version: ${{ matrix.go-version }}
+
+      - name: "Cache Go modules"
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/go/bin/*-v*
```
Comment (codefromthecrypt) [inline .github/workflows/commit.yaml:43]: ex

```bash
$ ls ~/go/bin/*-v*
/Users/adrian/go/bin/buf-v0.41.0           /Users/adrian/go/bin/golangci-lint-v1.39.0 /Users/adrian/go/bin/protoc-gen-go-v1.26.0
/Users/adrian/go/bin/goimports-v0.1.0      /Users/adrian/go/bin/licenser-v0.6.0       /Users/adrian/go/bin/shfmt-v3.2.4
```

File: .github/workflows/commit.yaml [hunk 3]
```diff
@@ -80,17 +89,16 @@ jobs:
...
-    runs-on: ${{ matrix.runner.os }}
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 90  # instead of 360 by default
     strategy:
       fail-fast: false  # don't fail fast as sometimes failures are operating system specific
       matrix:
-        runner:
-          - os: macos-latest
-          - os: ubuntu-latest
+        go-version: [1.16.3]
```
Comment (codefromthecrypt) [inline .github/workflows/commit.yaml:100]: sadly there's no top-level variable support in GH actions so we have to find/replace in this file (we also did before this change)

**Review by mathetake — APPROVED** [review]

**Pull request merged**